syntax = "proto3";

import "engine.proto";
import "google/protobuf/struct.proto"; // For Value

package com.wazuh.api.engine.test;

enum DebugMode
{
    ONLY_OUTPUT = 0;
    OUTPUT_AND_TRACES = 1;
    OUTPUT_AND_TRACES_WITH_DETAILS = 2;
}

message Run_Request
{
    optional string session = 1;
    optional google.protobuf.Value event = 2;
    optional uint32 protocolqueue = 3;
    optional DebugMode debugmode = 4;
    optional string protocollocation = 5;
}

message Run_Response
{
    ReturnStatus status = 1;                   // Status of the query
    optional string error = 2;                 // Error message if status is ERROR
    optional google.protobuf.Value output = 3; // test output
    optional google.protobuf.Value traces = 4; // test rules
}
